src/components/clusterprovider/minikube/minikube.ts

```diff
@@ -109,6 +133,41 @@ async function stopMinikube(context: Context): Promise<void> {
     });
 }
 
+async function minikubeStatus(context: Context, notification: boolean): Promise<MinikubeInfo> {
+    if (!await checkPresent(context, notification? CheckPresentMode.Alert : CheckPresentMode.Silent)) {
+        if (notification) {
+            return;
+        }
+        throw new Error('minikube executable could not be found!');
+    }
+
+    try {
+        const result = await context.shell.exec(
+            `${context.binPath} status --format '["{{.MinikubeStatus}}","{{.ClusterStatus}}","{{.KubeconfigStatus}}"]'`);
+        if (result.stderr.length === 0) {
+            const obj = JSON.parse(result.stdout);
+            if (notification) {
+                vscode.window.showInformationMessage(`Minikube is ${obj[0]}`);
+            }
+            return {
+                running: 'Stopped' !== obj[0],
+                cluster: obj[1],
+                kubectl: obj[2],
+            } as MinikubeInfo;
+        }
+        if (notification) {
+            vscode.window.showErrorMessage(`failed to get status: ${result.stderr}`);
```

Comment:
It would be good to clarify that this message refers to Minikube.  Also, I think we usually start messages with a capital letter.  E.g. `Failed to get Minikube status: ${…}` maybe?